pkg/validate/container_service.go

```diff
@@ -1,10 +1,258 @@
 package validate
 
 import (
-	acsapi "github.com/openshift/openshift-azure/pkg/api"
+	"fmt"
+	"regexp"
+
+	"github.com/openshift/openshift-azure/pkg/api"
 )
 
+var rxRfc1123 = regexp.MustCompile(`(?i)^` +
+	`([a-z0-9]|[a-z0-9][-a-z0-9]{0,61}[a-z0-9])` +
+	`(\.([a-z0-9]|[a-z0-9][-a-z0-9]{0,61}[a-z0-9]))*` +
+	`$`)
+
+var rxAgentPoolProfileVNetSubnetID = regexp.MustCompile(`(?i)^` +
+	`/subscriptions/[^/]+` +
+	`/resourceGroups/[^/]+` +
+	`/providers/Microsoft\.Network` +
+	`/virtualNetworks/[^/]+` +
+	`/subnets/[^/]+` +
+	`$`)
+
+var validAgentPoolProfileNames = map[string]struct{}{
+	string(api.AgentPoolProfileRoleCompute): struct{}{},
+	string(api.AgentPoolProfileRoleInfra):   struct{}{},
+	string(api.AgentPoolProfileRoleMaster):  struct{}{},
+}
+
+var validRouterProfileNames = map[string]struct{}{
+	"default": struct{}{},
+}
+
+func isValidHostname(h string) bool {
+	return len(h) <= 255 && rxRfc1123.MatchString(h)
+}
+
 // ContainerService validates a ContainerService struct
-func ContainerService(new, old *acsapi.ContainerService) (errs []error) {
+func ContainerService(new, old *api.ContainerService) (errs []error) {
```

Comment:
This should be updated to `OpenShiftCluster` or something similar.

Reply:
this is moving all validation to internal.  Internal data type is `ContainerService`.  We've talked about renaming it but I don't think it should be part of this PR